UI/DamageIndicator.cs

```diff
@@ -0,0 +1,44 @@
+using Godot;
+using System;
+
+public class DamageIndicator : TextureRect
+{
+    // Angle the bullet came from
+    Vector2 ShotFirePosition;
+    float OriginalAngle;
+    public override void _Ready()
+    {
+        // Start transparency Tween in _Ready so we know the Tween has been created
+        Tween TransparencyTween = GetNode<Tween>("Tween");
+        // Tween the "modulate" property
+        TransparencyTween.InterpolateProperty(this,"modulate", new Color(255,255,255,1), new Color(255,255,255,0), 3, Tween.TransitionType.Linear, Tween.EaseType.InOut);
+        TransparencyTween.Start();
+    }
+
+
+    public void SetTargetAngle(Vector2 ShotFirePosition)
```

Comment:
Function name does not match argument nor body

ShotFirePosition should be Vector3 and suffixed with Arg (C# explicitly allows variable shadowing only for arguments which name collide with a local member, SkyOfSteel convention is to avoid this by suffixing with Arg in these cases)